### PR TITLE
sqla extension enhencements

### PR DESCRIPTION
**Pull request by hadrien**
* Commit should not be done by sqla extension but in request finished callback
* Add a test on integrity error
* Reorganize sqla example and test
* Remove ``__repr__`` method on sqla base class
* ``sqla.Collection`` does not load entity when traversing by default.

### CODE REVIEW

File: example/ext/sqla/db/versions/1162611363e2_create_user_table.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ def upgrade():
     op.create_table(
         'project',
         sa.Column('id', sa.Integer, primary_key=True),
-        sa.Column('name', sa.String(50), nullable=False),
+        sa.Column('name', sa.String(50), nullable=False, unique=True),
```
Comment (hadrien) [inline example/ext/sqla/db/versions/1162611363e2_create_user_table.py]: To test an integrity error

File: royal/ext/sqla.py [hunk 2]
```diff
@@ -70,18 +67,8 @@ def __init__(self, name, parent, request, entity=None):
...
-        return '<%s item at %s named %r>' % (self.__class__.__name__,
-                                             id(self),
-                                             self.name)
-
-    def on_traversing(self, key):
-        self.load_entity()
-
     def load_entity(self):
         if self.entity is None:
-            if self.entity_cls is None:
```
Comment (deuxpi) [inline royal/ext/sqla.py]: Any reason why this verification is being removed?

File: royal/tests/functional/test_sa_example.py [hunk 3]
```diff
@@ -51,3 +53,8 @@ def test_project_update(self):
                     'name': u'new_name'}
         result = self.app.get('/projects/%s' % PR_ID).json
         self.assertEqual(result, expected)
+
+    def test_project_create_duplicate(self):
+
+        with self.assertRaises(IntegrityError):
```
Comment (deuxpi) [inline royal/tests/functional/test_sa_example.py]: `IntegrityError` in a **functional** test?

File: royal/ext/sqla.py (hunk 2 above)
Comment (hadrien) [inline royal/ext/sqla.py]: -_-  Not that I remember :\

**Pull request merged**